dev/NavigationView/NavigationView.cpp

```diff
@@ -2314,18 +2410,8 @@ void NavigationView::KeyboardFocusFirstItemFromItem(const winrt::NavigationViewI
 {
     auto const firstElement = [this, nvib]()
     {
-        if (IsTopNavigationView())
-        {
-            if (IsContainerInOverflow(nvib))
-            {
-                return m_topNavRepeaterOverflowView.get().TryGetElement(0);
-            }
-            else
-            {
-                return m_topNavRepeater.get().TryGetElement(0);
-            }
-        }
-        return m_leftNavRepeater.get().TryGetElement(0);
+        auto const parentIR = GetParentItemsRepeaterForContainer(nvib);
```

Comment:
In Hierarchical scenarios, this will put focus on the first item on the same level. I believe we want to put focus on the first menu item of the whole menu tree. Pinging @anawish@microsoft.com to confirm

Reply:
Sorry, just seeing this now, but yes that sounds correct.